### ISSUE TITLE
Questions for training gpt-2 using mup

### ISSUE THREAD

**Issue by jiangjiadi** [microsoft/mup#66]
The conclusion of the paper is engaging. But when I tried to implement hyperparameter transfer using Mup on the GPT-2 model, I encountered some issues.
- When scaling the width of the GPT-2 model, I found the optimum shifts still existed, even though I used Mup Initialization. Some of the results are as follows. (Our code is modified from [karpathy/nanoGPT: The simplest, fastest repository for training/finetuning medium-sized GPTs. (github.com)](https://github.com/karpathy/nanoGPT/tree/master)) I have checked the std for each layer to ensure the implementation of Mup was not problematic. To eliminate the optimum shifts in mup, I also made some attempts,  including whether to scale the weight decay, trying different base shape, and whether to using mixed precision. But these attempts still haven't worked. Can you provide me with more implementation details of Mup on the GPT model?

| lr | 96 | 192 | 384 | 768 |
| ---- | ---- | ---- | ---- | ---- |
|  2^-8|	4.1291|	3.7287|	3.4713|	nan|
| 2^-9|	**4.1184**|	3.7086|	3.4041|	
| 2^-10|	4.1627|	**3.7074**|	3.3809|	3.1226|
|2^-11|	4.2398|	3.7363|	**3.3776**|	**3.1009**|
|2^-12|	4.3393|	3.8111|	3.4135|	3.1074|

- I don't quite understand the attention logit scaling mentioned in the article. If the attention logit is not scaled, would Mup become ineffective?

**Comment by edwardjhu** [comment]
Can you share the coordinate check plots like these?

https://github.com/microsoft/mup/tree/main/examples/Transformer/coord_checks

They are really helpful for debugging.

**Comment by jiangjiadi** [comment]
The coordinate check plots are given below. It seems the Mup's plot is much more stable than SP's. However, in the case of high learning rate, sometimes there will be some fluctuations in Mup's plot, and I am not sure whether these fluctuations have an impact on Mup.
- SP, Adamw, lr=0.001
![image](https://github.com/microsoft/mup/assets/34134495/e6357997-f95d-48b7-803e-9a389429f8b7)
- SP, Adamw, lr=0.01
![image](https://github.com/microsoft/mup/assets/34134495/25198f6b-d0eb-461d-9231-ed6fd8010a8e)
- SP, Adamw, lr=0.1
![image](https://github.com/microsoft/mup/assets/34134495/62076583-7089-4dfd-9874-f1b46365ca34)

- Mup, Adamw, lr=0.001
![image](https://github.com/microsoft/mup/assets/34134495/33a23be3-9fb5-4be3-b2f5-67c565a0894f)
- Mup, Adamw, lr=0.01
![image](https://github.com/microsoft/mup/assets/34134495/dee9aef0-2151-42bd-900e-27e53ac00118)
- Mup, Adamw, lr=0.1
![image](https://github.com/microsoft/mup/assets/34134495/a5522afd-7488-4d03-96cf-a09272c9727a)

**Comment by edwardjhu** [comment]
Can you extend the x-axis to the right as much as you can for mup?

**Comment by jiangjiadi** [comment]
> Can you extend the x-axis to the right as much as you can for mup?

Now, I have tested the width from 48 to 3072 with the base shape = 768.
- SP, adamw, lr=0.001
![image](https://github.com/microsoft/mup/assets/34134495/06091874-99bf-48fa-b0f3-abc43a2dc5b7)

- Mup, adamw, lr=0.001
![image](https://github.com/microsoft/mup/assets/34134495/2c3a94d6-13ab-4958-ab74-dd8ab103e8f9)

**Comment by jiangjiadi** [comment]
I've identified the problem. Thanks~

**Comment by Joelx** [comment]
@jiangjiadi Could you share what the problem was and how you remedied it?
